src/PetriEngine/Reducer.cpp

```diff
@@ -1523,15 +1523,16 @@ namespace PetriEngine {
 
         bool continueReductions = false;
         for (size_t t1 = 0; t1 < parent->numberOfTransitions() - 1; ++t1) {
+            // We check if t1 can be removed
+            Transition &tran1 = getTransition(t1);
+            if (tran1.skip || tran1.inhib) continue;
+
             for (size_t t2 = t1 + 1; t2 < parent->numberOfTransitions(); ++t2) {
                 if (hasTimedout()) return false;
 
-                // We check if t1 or t2 can be removed
-                Transition &tran1 = getTransition(t1);
-                if (tran1.skip || tran1.inhib) break;
-
+                // We check if t2 can be removed
                 Transition &tran2 = getTransition(t2);
-                if (tran2.skip || tran2.inhib) continue;
```

Comment:
Er det ok hvis vi ender med at swap nede ved linje ~1574

Reply:
God pointe, tænker lige over det. Skal måske også lidt mere logik ind for at understøtte inhibitor arcs faktisk. Er vidst også tran1 der gerne må være inhibited her, da det er den vi antager vi kan fjerne fra starten. In conclusion: ignore this PR for now